### ISSUE TITLE
Format a number as currency using a compact notation

### ISSUE THREAD

**Issue by itingliu** [apple/swift-foundation#379]
Reported on forum: https://forums.swift.org/t/format-currency-using-a-compact-notation/69443

We should support `notation` for format styles that formats the number as a currency. i.e. one should be able to do this:

```swift
let price = Decimal(1_500_000.59)
price.formatted(.currency(code: "USD").notation(.compactName)) 
```
The above should return "$1.5M" as opposed to "$1,500,000.59"

**Comment by kylesluder** [comment]
Since “billion” is [potentially ambiguous](https://en.wikipedia.org/wiki/1,000,000,000)—it can mean either 1,000 million or 1 million million—should million be the largest scale? Wikipedia implies that the long scale has gone out of use, but I have personally seen confusion over short and long scale billions in recent financial reporting.

For Indian English locales, should compact notation use lakh and crore instead of millions?

**Comment by itingliu** [comment]
I think the expected behavior is that the number part would be exactly the same as that of the "plain number" format style, i.e. `compactNumber` in this example
```swift
let price = Decimal(1_500_000.59)
let compactNumber = price.formatted(.number.notation(.compactName)) 
```

**Comment by itingliu** [comment]
This proposal aims to address this:
https://github.com/apple/swift-foundation/blob/main/Proposals/0008-notation-formatting-for-currencies.md

**Comment by itingliu** [comment]
This is now available thanks to #449
